### PR TITLE
X-Forwarded-For catcher

### LINKED ISSUE #1714: unable to process remote addr in auth/reauth
Thanks @Firstyear! Sorry to keep bugging you. I just pulled the latest build, and I think there might be a small bug. This is what I'm seeing:
```
7ae454b3-4d25-4236-93bd-67ba26b60426 INFO     tide-request [ 103µs | 100.00% ]
7ae454b3-4d25-4236-93bd-67ba26b60426 INFO     ┝━ ｉ [info]: Request received | event_tag_id: 9 | remote_addr: "72.24.19.178" | http.host: "idm.example.com" | http.method: "POST" | path: "/v1/auth"
7ae454b3-4d25-4236-93bd-67ba26b60426 ERROR    ┝━ 🚨 [error]: Unable to process remote addr, refusing to proceed
7ae454b3-4d25-4236-93bd-67ba26b60426 ERROR    ┕━ 🚨 [error]: Internal error -> Response sent | event_tag_id: 4 | message: unable to validate peer address | error_type: "?" | status: 500 - Internal Server Error
```

Which I think is [here](https://github.com/kanidm/kanidm/pull/1711/files#diff-fb54b766a3ec8982eedf95e36742687d4532625889784056c547dafdd03bd256R1102) in the PR. I don't think it's my nginx conf since the log above shows the expected IP address. I think it's just not always getting passed around properly?

Also, would you like me to make a proper issue ticket for this, or is it okay to have this conversation here?

_Originally posted by @dlrobson in https://github.com/kanidm/kanidm/issues/1711#issuecomment-1584474324_

### PR DESCRIPTION
**Pull request by yaleman**
Fixes #1714 - thanks to @dlrobson for picking it up!

Also:

- spelling thing
- minor formatting in the docs
- minor formatting thing in Cargo.toml which was making the TOML parser upset
- I can't remember what else I did

Checklist

- [x] This pr contains no AI generated code
- [x] cargo fmt has been run
- [x] cargo clippy has been run and it's not my code I swear
- [x] cargo test has been run and passes

### CODE REVIEW

**Review by Firstyear — APPROVED** [review]

File: server/core/src/https/mod.rs [hunk 1]
```diff
@@ -183,14 +183,25 @@ impl RequestExtensions for tide::Request<AppState> {
...
 
+    /// Returns the remote address of the client, based on if you've got trust_x_forward_for set in config.
     fn get_remote_addr(&self) -> Option<IpAddr> {
         if self.state().trust_x_forward_for {
-            self.remote()
+            // split the socket address off if we've got one, then parse it as an `IpAddr`
+            // xff headers don't have a port, but if we're going direct you might get one
+            let res = self
+                .remote()
+                .map(|addr| addr.split(':').next().unwrap_or(addr))
```
Comment (mweinelt) [inline server/core/src/https/mod.rs:193]: I'm pretty sure that discounts the existence of IPv6 addresses.

```
 kanidmd[166700]: ed6a5f2a-56be-479a-9315-82b6dcf608c4 INFO     tide-request [ 56.7µs | 100.00% ]
 kanidmd[166700]: ed6a5f2a-56be-479a-9315-82b6dcf608c4 INFO     ┝━ ｉ [info]: Request received | event_tag_id: 9 | remote_addr: "2a01:4f8:c2c:e698::1" | http.host: "accounts-test.darmstadt.ccc.de" | http.method: "POST" | path: "/v1/auth"
 kanidmd[166700]: ed6a5f2a-56be-479a-9315-82b6dcf608c4 ERROR    ┝━ 🚨 [error]: Unable to get remote addr for auth event, refusing to proceed
 kanidmd[166700]: ed6a5f2a-56be-479a-9315-82b6dcf608c4 ERROR    ┕━ 🚨 [error]: Internal error -> Response sent | event_tag_id: 4 | message: unable to validate peer address | error_type: "?" | status: 500 - Internal Server Error
```

Comment (mweinelt) [reply server/core/src/https/mod.rs:193]: Looks like a job for `SocketAddr::parse_ascii` or similar.

Comment (yaleman) [reply server/core/src/https/mod.rs:193]: Yeah, I realised last night about the IPv6 thing 😞

**Comment by Firstyear** [comment]
@yaleman I think we just have to chain ipAddr::parse().or_else(socket_addr::parse). It's the most reliable solution IMO.

**Comment by yaleman** [comment]
> @yaleman I think we just have to chain ipAddr::parse().or_else(socket_addr::parse). It's the most reliable solution IMO.

Yep, that's what I was thinking.

**Comment by Firstyear** [comment]
Done #1729

**Pull request merged**